hera_cal/io.py

```diff
@@ -864,6 +864,147 @@ def iterate_over_times(self, Nints=1, times=None):
             yield self.read(times=times[i:i + Nints])
 
 
+def read_hera_hdf5(filenames, bls=None, pols=None, full_read_thresh=0.002,
+                   read_data=True, read_flags=False, read_nsamples=False, check=False,
+                   dtype=np.complex128, verbose=False):
+    '''A ~100x faster interface to getting data out of HERA HDF5 files. Only concatenates
+    along time axis. Assumes that if antenna_numbers header array stays the same,
+    the baseline order in the files stays the same. Puts times in ascending order,
+    but does not check that files are contiguous.
+
+    Arguments:
+        filenames: list of files to read
+        bls: list of (ant_1, ant_2, [polstr]) tuples to read out of files
+        pols: list of pol strings to read out of files
+        full_read_thresh (0.01): fractional threshold for reading whole file 
+                                 instead of baseline by baseline.
+        read_data (bool, True): read visdata
+        read_flags (bool, False): read flags
+        read_nsamples (bool, False): read nsamples
+        check (bool, False): run sanity checks to make sure files match.
+        dtype (np.complex128): numpy datatype for output complex-valued arrays
+        verbose: print some progress messages.
+
+    Returns:
+        rv: dictionary with keys 'info' (metadata), 'visdata' (dictionary of waterfalls
```

Comment:
In most of hera_cal, `visdata` is just called `data`. Any objections to calling it that?

Reply:
Also, I kind of feel like this would slightly easier to use if it returned 3 or 4 arguments and just stuck `None` if you didn't ask for nsamples or flags or whatever. That's a bit more similar to how `HERAData.read()` and `HERACal.read()` work.

Reply:
This is how it was originally, but then I added options for reading data/flags/nsamples in any subset, and I didn't want a case where positional returns got misinterpreted.

I agree most of hera_cal calls it "data", but pyuvdata (and the file itself) calls it "visdata", and I think of this function as a thin stand-in to pyuvdata and would like it to stay as close to the file format as possible

Reply:
Meh, I'll do 'data'.